src/digraph6.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{get_edge_iter_with_weights, StablePyGraph};
+use crate::graph6::{utils, IOError, GraphConversion};
+use pyo3::prelude::*;
+use pyo3::types::PyAny;
+use petgraph::graph::NodeIndex;
+use petgraph::algo;
+
+/// Directed graph implementation (extracted from graph6.rs)
+#[derive(Debug)]
+pub struct DiGraph {
+    pub bit_vec: Vec<usize>,
+    pub n: usize,
+}
+impl DiGraph {
+    /// Creates a new DiGraph from a graph6 representation string
+    pub fn from_d6(repr: &str) -> Result<Self, IOError> {
+        let bytes = repr.as_bytes();
+        Self::valid_digraph(bytes)?;
+    let (n, n_len) = utils::parse_size(bytes, 1)?;
+    let Some(bit_vec) = Self::build_bitvector(bytes, n, 1 + n_len) else {
+            return Err(IOError::NonCanonicalEncoding);
+        };
+        Ok(Self { bit_vec, n })
+    }
+
+    /// Creates a new DiGraph from a flattened adjacency matrix
+    #[cfg(test)]
```

Comment:
We don't run tests for `rustworkx` with `cargo test`. Please add an equivalent test that covers this in `.py`